vcd/resource_vcd_network_routed.go

```diff
@@ -444,26 +511,42 @@ func resourceVcdNetworkIPAddressHash(v interface{}) int {
 // On success, returns the name of the edge gateway
 func findEdgeGatewayConnection(client *VCDClient, vdc *govcd.Vdc, network *govcd.OrgVDCNetwork) (string, error) {
```

Comment:
The name of this function is very confusing. I was reading the big comment at:
https://github.com/terraform-providers/terraform-provider-vcd/pull/402/files#diff-e99542a876fc291b70cf88bd368423b5R41-R48

It just didn't make any sense, then I looked at this function and realized it's not returning a "Connection". Better name needed.

Reply:
The name says what the function is doing: finding a connection between a routed network and an edge gateway. Please suggest a better name if you have one.

Reply:
More background info:
when we create a routed network, we know the name of the edge gateway. When we retrieve a network for importing into a state file, we don't know the edge gateway name, as it is not needed to retrieve the network. Thus, once we have the network, we need to find to which edge gateway is connected. This function exists since the creation of the network data source. It was using a different algorithm then, by looking at the existing edge gateways for a connected network. However, for distributed networks the edge gateway doesn't store a connection, and we had to change the method of connection retrieval using a query.

Reply:
Makes it clearer. Assuming I got the purpose right, then the name suggestion is:
`findEdgeGatewayNameByNetwork`
This would be clearer and consistent with our format of `GetEdgeGatewayByName`

Finally, I just realized that this function is in provider, we should really move it to govcd.

Reply:
One more variant:
`findEdgeGatewayNameOfNetwork`

Reply:
Moved to govcd